snowplow-demo-app/src/androidTest/java/com/snowplowanalytics/snowplowtrackerdemo/context/ApplicationContextTest.java

```diff
@@ -0,0 +1,67 @@
+package com.snowplowanalytics.snowplowtrackerdemo.context;
+
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Build;
+
+import com.snowplowanalytics.snowplow.tracker.constants.Parameters;
+import com.snowplowanalytics.snowplow.tracker.payload.SelfDescribingJson;
+import com.snowplowanalytics.snowplow.tracker.tracker.InstallTracker;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.filters.SmallTest;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import static com.snowplowanalytics.snowplow.tracker.utils.Util.addToMap;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+@SmallTest
+public class ApplicationContextTest {
+
+    private Context context;
+
+    @Before
+    public void setUp() {
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+    }
+
+    @Test
+    public void testGetApplicationContext() {
```

Comment:
Can we call the function itself instead of rewriting it here?

Reply:
It looks like a rewrite but `getApplicationContext` is called on line 57. We need to fetch `versionName` and `versionCode` so that we can compare result of `getApplicationContext` with the fetched values. Do I miss something?